tests/components/homematicip_cloud/test_config_flow.py

```diff
@@ -76,92 +108,104 @@
         const.HMIPC_PIN: "123",
         const.HMIPC_NAME: "hmip",
     }
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
 
-    with patch.object(
-        hmipc.HomematicipAuth, "async_setup", return_value=mock_coro(True)
-    ), patch.object(
-        hmipc.HomematicipAuth, "async_checkbutton", return_value=mock_coro(False)
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=False,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_setup",
+        return_value=True,
     ):
-        result = await flow.async_step_init(user_input=config)
-        assert result["type"] == "form"
-        assert result["errors"] == {"base": "press_the_button"}
-
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "user"}, data=config
+        )
 
-async def test_init_flow_show_form(hass):
-    """Test config flow shows up with a form."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
-
-    result = await flow.async_step_init(user_input=None)
     assert result["type"] == "form"
+    assert result["step_id"] == "link"
+    assert result["errors"] == {"base": "press_the_button"}
 
 
-async def test_init_flow_user_show_form(hass):
+async def test_init_flow_show_form(hass):
     """Test config flow shows up with a form."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
 
-    result = await flow.async_step_user(user_input=None)
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "user"}
+    )
     assert result["type"] == "form"
+    assert result["step_id"] == "init"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)